test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -362,7 +362,53 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
 			Assert.fail("should throw error. Class does not exist");
 		} catch (NoClassDefFoundError e) {}
 	}
-	
+
+	/*
+	 * Test ifacmp on value class
+	 * 
+	 * class TestIfacmpOnValueClass {
+	 *  
+	 * }
+	 */
+	@Test(priority=1)
+	static public void TestIfacmpOnValueClass() throws Throwable {
+		String fields[] = {"x:I", "y:I"};
+		point2DClass = ValueTypeGenerator.generateValueClass("Point2DInAcmp", fields);
+		makePoint2D = lookup.findStatic(point2DClass, "makeValue", MethodType.methodType(point2DClass, int.class, int.class));
+
+		int x = 0;
+		int y = 0;
+		
+		Object point2D = makePoint2D.invoke(x, y);
+
+		Object normalClass = (Object) x;
+		/*
+         * Test if bytecode if_acmpne behaves correctly
+         */
+		if (point2D == normalClass) {
+		    Assert.fail("should throw error. Acmpne passes one valuetype and one normal object!");
```

Comment:
you can modify comment to say, "Equality comparisons with valueTypes should always return false"